### PR TITLE
Skip 'extra_tests_phub' test if packagehub is not ready

### PR DESCRIPTION
**Pull request by rfan1**
- Related ticket:https://progress.opensuse.org/issues/120879
- Verification run: https://openqa.suse.de/tests/10041939
[Please ignore the failed cases, they are known issues]

### CODE REVIEW

File: tests/console/add_phub_extension.pm [hunk 1]
```diff
@@ -16,13 +16,21 @@ use registration qw(add_suseconnect_product get_addon_fullname);
...
 
 # Add milestone flag to save setup into the snapshot which
 # packagehub is activated
 sub test_flags {
-    return {milestone => 1};
+    return {milestone => 1, fatal => 1};
 }
 
-1;
+;
```
Comment (coolgw) [inline tests/console/add_phub_extension.pm:36]: do not remove this

File: tests/console/add_phub_extension.pm [hunk 2]
```diff
@@ -16,13 +16,22 @@ use registration qw(add_suseconnect_product get_addon_fullname);
 sub run {
     select_serial_terminal;
 
-    add_suseconnect_product(get_addon_fullname('phub'));
+    eval { add_suseconnect_product(get_addon_fullname('phub')); };
+    if ($@) {
+        if (check_var('BETA', '1')) {
+            force_soft_failure('poo#120879, PackageHub installation might fail in early development');
+            set_var('PHUB_READY', 'no');
```
Comment (foursixnine) [inline tests/console/add_phub_extension.pm:23]: Please add this variable in Variables.md

Comment (foursixnine) [inline tests/console/add_phub_extension.pm:23]:
```suggestion
            set_var('PHUB_READY', 0);
```

Follows more what we have?

File: tests/console/python_scientific.pm [hunk 3]
```diff
@@ -37,6 +37,8 @@ sub run_python_script {
 sub run {
     select_serial_terminal;
 
+    return if (check_var('PHUB_READY', 'no'));
```
Comment (foursixnine) [inline tests/console/python_scientific.pm:40]: Maybe if you extract the variable to either `registration` we can do:

```suggestion
    return if is_phub_ready();
```

which is cleaner :)

**Review by foursixnine — COMMENTED** [review]
Looking good!

File: tests/console/python_scientific.pm (hunk 3 above)
Comment (rfan1) [reply tests/console/python_scientific.pm:40]: You are correct, let me do that

**Review by pdostal — CHANGES_REQUESTED** [review]
Interesting solution. We have some `phub` tests also in QE-C. cc: @jlausuch @grisu48 
Also, is thae `PHUB_READY` new variable? If so it needs to be listed in `variables.md`.

**Comment by rfan1** [comment]
> Interesting solution. We have some `phub` tests also in QE-C. cc: @jlausuch @grisu48 Also, is thae `PHUB_READY` new variable? If so it needs to be listed in `variables.md`.

Thanks, already there:
PHUB_READY | boolean | true | Indicates PackageHub is available, it may be not ready in early development phase[Before Beta].

File: lib/registration.pm [hunk 4]
```diff
@@ -104,6 +105,14 @@ sub is_module {
     return defined $SLE15_MODULES{$name};
 }
 
+# Check if Packagehub is available
+sub is_phub_ready {
+    if (check_var('PHUB_READY', '0')) {
+        return 0;
+    }
+    return 1;
```
Comment (pdostal) [inline lib/registration.pm:113]:
```suggestion
    return (check_var('PHUB_READY', '0') ? 0 : 1;
```

File: tests/console/add_phub_extension.pm [hunk 5]
```diff
@@ -11,18 +11,29 @@ use strict;
...
+    eval { add_suseconnect_product(get_addon_fullname('phub')); };
+    if ($@) {
+        if (check_var('BETA', '1')) {
+            force_soft_failure('poo#120879, PackageHub installation might fail in early development');
+            set_var('PHUB_READY', 0);
+        }
+        else {
+            die "$@";
+        }
+    }
```
Comment (pdostal) [inline tests/console/add_phub_extension.pm:30]: I don't understand this.

**Review by pdostal — DISMISSED** [review]

**Comment by jlausuch** [comment]
We have always used the condition `BETA=0|1`  to run packagehub tests.

File: tests/console/add_phub_extension.pm (hunk 5 above)
Comment (rfan1) [reply tests/console/add_phub_extension.pm:30]: Hmm,  I will have a try to activated phub, if it fails with `BETA=1`, then force fail the test

**Comment by rfan1** [comment]
> We have always used the condition `BETA=0|1` to run packagehub tests.

Yes, I agree with you on this point, as we don't need to cover the PackageHub tests before GMC.
However, based on https://bugzilla.suse.com/show_bug.cgi?id=1202416, the PackageHub should be ready in Beta phase, Is it too late for us to carry out the tests in GMC phase?

File: tests/console/python_scientific.pm [hunk 6]
```diff
@@ -37,6 +38,8 @@ sub run_python_script {
 sub run {
     select_serial_terminal;
 
+    return unless is_phub_ready();
```
[the remark this answers is not in the document]
Comment (rfan1) [reply tests/console/python_scientific.pm:41]: sure

**Comment by jlausuch** [comment]
> > We have always used the condition `BETA=0|1` to run packagehub tests.
> 
> Yes, I agree with you on this point, as we don't need to cover the PackageHub tests before GMC. However, based on https://bugzilla.suse.com/show_bug.cgi?id=1202416, the PackageHub should be ready in Beta phase, Is it too late for us to carry out the tests in GMC phase?

Ok, since we have that issue repeatably in every release, I agree to create a new variable PH READY. Thanks!

**Comment by rfan1** [comment]
@pdostal @foursixnine Could you please help review again, I have revised my codes as your kindly suggestion.

**Pull request merged**